src/kernel.cu

```diff
@@ -103,3 +121,83 @@ void generateIPv6PacketsKernel(int numPackets, bool debug) {
     // Free the memory for IPv6 packets on the host
     delete[] h_ipv6Packets;
 }
+
+uint32_t generateRandomIPv4AddressHost() {
+  return rand() % 0xFFFFFFFF; // Generate and return a random IPv4 address
+}
+
+uint8_t* generateRandomIPv6AddressHost() {
+  uint8_t* address = new uint8_t[16]; // Allocate memory for the address
+
+
+  for (int i = 0; i < 16; ++i) {
+    address[i] = static_cast<uint8_t>(std::rand() % 256); // Generate a random byte for each octet
+  }
+  return address; // Return the generated address
+}
+void createRoutingTable(GlobalPacketData& globalPacketData) {
```

Comment:
Add a bool debug arg. Print statements should be executed only when this debug is set.